### PR TITLE
Filter-scenarios based on categories

### PR DESCRIPTION
**Pull request by abrahimi93**
closes https://github.com/hobbyfarm/hobbyfarm/issues/232

### CODE REVIEW

File: src/app/scenario/scenario.component.html [hunk 1]
```diff
@@ -29,10 +29,28 @@ <h3>Scenarios</h3>
...
-    <clr-datagrid [clrDgSingleSelected]="unusedSelectedScenario" (clrDgSingleSelectedChange)="editScenario($event)" [clrDisableSelection]="!selectRbac">
+
+    <clr-dropdown>
+      <button class="dropdown-toggle" clrDropdownTrigger>
+        <span *ngIf="category==''">All</span>
+        <span *ngIf="category=='__none__'">No Category</span>
+        <span *ngIf="category !='' && category != '__none__' ">{{category}}</span>
+        <clr-icon shape="caret down"></clr-icon>
+      </button>
+      <clr-dropdown-menu clrPosition="top-left" *clrIfOpen>
```
Comment (PhPrinz) [inline src/app/scenario/scenario.component.html:40]:
```suggestion
      <clr-dropdown-menu clrPosition="bottom-left" *clrIfOpen>
```

If the List of Categories has some more entries, the Dropdown List gets hidden behind the frame of the Browser Window

File: src/app/scenario/scenario.component.html [hunk 2]
```diff
@@ -29,10 +29,28 @@ <h3>Scenarios</h3>
 </div>
 <div class="clr-row">
   <div class="clr-col-4">
-    <clr-datagrid [clrDgSingleSelected]="unusedSelectedScenario" (clrDgSingleSelectedChange)="editScenario($event)" [clrDisableSelection]="!selectRbac">
+
+    <clr-dropdown>
+      <button class="dropdown-toggle" clrDropdownTrigger>
+        <span *ngIf="category==''">All</span>
```
Comment (PhPrinz) [inline src/app/scenario/scenario.component.html:35]: After selecting "All" or "No Category" the dropdown doesn't close by itself as it does with the other selectable categories. Tested with Firefox and Chrome.

File: src/app/scenario/scenario.component.ts [hunk 3]
```diff
@@ -387,6 +394,49 @@ export class ScenarioComponent implements OnInit {
...
+    this.filterScenarioList();
+  }
+
+  calculateCategories() {
+    let c = new Set<string>();
+    this.scenarios.forEach(s => {
+      console.log(s);
+      s.categories.forEach(category => c.add(category));
+    });
+    console.log(c);
```
Comment (PhPrinz) [inline src/app/scenario/scenario.component.ts:436]:
```suggestion
```

File: src/app/scenario/scenario.component.ts [hunk 4]
```diff
@@ -387,6 +394,49 @@ export class ScenarioComponent implements OnInit {
...
+
+  setCategoryFilter(category: string) {
+    this.category = category
+    this.filterScenarioList();
+  }
+
+  calculateCategories() {
+    let c = new Set<string>();
+    this.scenarios.forEach(s => {
+      console.log(s);
```
Comment (PhPrinz) [inline src/app/scenario/scenario.component.ts:433]:
```suggestion
```

**Review by PhPrinz — CHANGES_REQUESTED** [review]

File: src/app/scenario/scenario.component.html [hunk 5]
```diff
@@ -29,10 +29,28 @@ <h3>Scenarios</h3>
...
+        <span *ngIf="category=='__none__'">No Category</span>
+        <span *ngIf="category !='' && category != '__none__' ">{{category}}</span>
+        <clr-icon shape="caret down"></clr-icon>
+      </button>
+      <clr-dropdown-menu clrPosition="top-left" *clrIfOpen>
+        <div class="dropdown-item" (click)="setCategoryFilter('')">All</div>
```
Comment (jggoebel) [inline src/app/scenario/scenario.component.html:41]: add 'clrDropdownItem' to close dropown after selection. Add to no Category Filter aswell

File: src/app/scenario/scenario.component.html [hunk 6]
```diff
@@ -29,10 +29,28 @@ <h3>Scenarios</h3>
...
-      <clr-dg-row *clrDgItems="let s of scenarios" [clrDgItem]="s">
+    <clr-dropdown>
+      <button class="dropdown-toggle" clrDropdownTrigger>
+        <span *ngIf="category==''">All</span>
+        <span *ngIf="category=='__none__'">No Category</span>
+        <span *ngIf="category !='' && category != '__none__' ">{{category}}</span>
+        <clr-icon shape="caret down"></clr-icon>
+      </button>
+      <clr-dropdown-menu clrPosition="bottom-left" *clrIfOpen>
+        <div class="dropdown-item" (click)="setCategoryFilter('')" clrDropdownItem>select category</div>
```
Comment (jggoebel) [inline src/app/scenario/scenario.component.html:41]: "All categories" OR "All scenarios"

File: src/app/scenario/scenario.component.html [hunk 7]
```diff
@@ -29,10 +29,28 @@ <h3>Scenarios</h3>
...
 <div class="clr-row">
   <div class="clr-col-4">
-    <clr-datagrid [clrDgSingleSelected]="unusedSelectedScenario" (clrDgSingleSelectedChange)="editScenario($event)" [clrDisableSelection]="!selectRbac">
-      <clr-dg-column>Name</clr-dg-column>
 
-      <clr-dg-row *clrDgItems="let s of scenarios" [clrDgItem]="s">
+    <clr-dropdown>
+      <button class="dropdown-toggle" clrDropdownTrigger>
+        <span *ngIf="category==''">All</span>
+        <span *ngIf="category=='__none__'">No Category</span>
```
Comment (jggoebel) [inline src/app/scenario/scenario.component.html:36]: "Without category"

File: src/app/scenario/scenario.component.html [hunk 8]
```diff
@@ -29,10 +29,28 @@ <h3>Scenarios</h3>
 </div>
 <div class="clr-row">
   <div class="clr-col-4">
-    <clr-datagrid [clrDgSingleSelected]="unusedSelectedScenario" (clrDgSingleSelectedChange)="editScenario($event)" [clrDisableSelection]="!selectRbac">
-      <clr-dg-column>Name</clr-dg-column>
 
-      <clr-dg-row *clrDgItems="let s of scenarios" [clrDgItem]="s">
+    <clr-dropdown>
+      <button class="dropdown-toggle" clrDropdownTrigger>
+        <span *ngIf="category==''">All</span>
```
Comment (jggoebel) [inline src/app/scenario/scenario.component.html:35]: "All scenarios" OR "All categories"

File: src/app/scenario/scenario.component.html [hunk 9]
```diff
@@ -29,10 +29,28 @@ <h3>Scenarios</h3>
...
+        <span *ngIf="category=='__none__'">Without Category</span>
+        <span *ngIf="category !='' && category != '__none__' ">{{category}}</span>
+        <clr-icon shape="caret down"></clr-icon>
+      </button>
+      <clr-dropdown-menu clrPosition="bottom-left" *clrIfOpen>
+        <div class="dropdown-item" (click)="setCategoryFilter('')" clrDropdownItem>All Categories</div>
+        <div class="dropdown-item" (click)="setCategoryFilter('__none__')" clrDropdownItem>Without Category</div>
+        <div class="dropdown-divider" role="separator" aria-hidden="true"></div>
+        <div class="dropdown-item" *ngFor="let c of categories" (click)="setCategoryFilter(c)" clrDropdownItem>{{c}}
+          clrDropdownItem
```
Comment (jggoebel) [inline src/app/scenario/scenario.component.html:45]: duplicate

**Pull request merged**